### PR TITLE
Add support for registering temporary stores

### PR DESCRIPTION
**Pull request by MaxRandom**
Temporary store - store registered in dispatcher, but not stored in domain. Lifecycle is bind to specific class(viewModel, service, controller etc.)

To create temporary store you need:
1. Create store instance.
2. Make strong reference to it.
3. Register store in dispatcher through `- (void)registerTemporaryStore:(TEBaseStore *)store` method.
4. Use as common store

### CODE REVIEW

File: Pod/Classes/TEDomain.m [hunk 1]
```diff
@@ -94,6 +94,16 @@ - (void)sendActionToMiddlewares:(TEBaseAction *)action
     }];
 }
 
+- (void)registerTemporaryStore:(TEBaseStore *)store {
```
Comment (sergeyzenchenko) [inline Pod/Classes/TEDomain.m:97]: copy paste from registerStore some code can be reused.

Comment (MaxRandom) [reply Pod/Classes/TEDomain.m:97]: I suggest to create `- (void)_registerStore:(TEBaseStore *)store` internal method that will register store in midlewares and dispatcher
Or `subscribeToEvents:`
@sergeyzenchenko what do you think?

Comment (sergeyzenchenko) [reply Pod/Classes/TEDomain.m:97]: you can have registerStore method and registerTemporaryStore and registerGlobalStore

**Comment by goncharik** [comment]
@MaxRandom why did travis build fail?

File: Pod/Classes/Dispatcher/TEDispatcherProtocol.h [hunk 2]
```diff
@@ -8,9 +8,12 @@
 
 #import <Foundation/Foundation.h>
 @class TEBaseAction;
+@class TEBaseStore;
 
 @protocol TEDispatcherProtocol <NSObject>
 
+@property (nonatomic, weak) TEBaseStore *store;
```
Comment (goncharik) [inline Pod/Classes/Dispatcher/TEDispatcherProtocol.h]: shouldn't it be readonly?

File: Pod/Classes/Dispatcher/TEStoreDispatcher.m [hunk 3]
```diff
@@ -15,12 +15,13 @@
 @interface TEStoreDispatcher ()
 
 @property (nonatomic, strong) NSMutableDictionary *callbacks;
-@property (nonatomic, weak) TEBaseStore * store;
 
 @end
 
 @implementation TEStoreDispatcher
 
+@synthesize store = _store;
```
Comment (goncharik) [inline Pod/Classes/Dispatcher/TEStoreDispatcher.m]: why do we need `synthesize` here?

**Comment by MaxRandom** [comment]
🎉

**Pull request merged**